### PR TITLE
Seeding

### PR DESCRIPTION
**Pull request by chrsbell**
Implemented populating database with random reviews on server initialization.

### CODE REVIEW

File: server/db.js [hunk 1]
```diff
@@ -14,85 +14,94 @@ const config = {
...
-    accuracy: Number,
-    location: Number,
-    value: Number
-  }
-});
-
-db.on('error', console.error.bind(console, 'connection error:'));
-db.once('open', function() {
-  // populate the database once we are connected
-  initDB();
```
Comment (chrisalopez1337) [inline server/db.js:37]: Dont call initDB() in the open, as it will be called more than once and create too much data.

File: server/db.js [hunk 2]
```diff
@@ -14,85 +14,94 @@ const config = {
...
                       'Joel', 'Johnny', 'Joesph', 'Karl', 'Katharine', 'Liz', 'Mike', 'Minji', 'Mylani', 'Rebecca',
                       'Rob', 'Shaquon', 'Sophie', 'Sokhary', 'Susan', 'Victoria', 'Watson', 'Yas'];
 
-const random_text = ['apple', 'orange', 'pear', 'little', 'big', 'hack', 'reactor', 'review', 'great', 'the', 'a',
-                     'is', 'of', 'will', 'tree', 'normal', 'abstract', 'your', 'job', 'free', 'work', 'stay','home'.
+const randomText = ['apple', 'orange', 'pear', 'little', 'big', 'hack', 'reactor', 'review', 'great', 'the', 'a',
+                     'is', 'of', 'will', 'tree', 'normal', 'abstract', 'your', 'job', 'free', 'work', 'stay','home',
                      'apartment', 'kite', 'rent', 'stay', 'like', 'roof', 'room', 'bathroom', 'lorem ipsum'];
 
 const generateRandomReview = (length) => {
```
Comment (chrisalopez1337) [inline server/db.js]: Dont need to pass length in as an arg, as you have already written a getRandomInt function that can be used inside of this function to generate a random length of a review.

File: server/db.js [hunk 3]
```diff
@@ -14,85 +14,94 @@ const config = {
...
-  let review = [];
+  let review = '';
```
Comment (chrisalopez1337) [inline server/db.js]: Can leave review as an arr, and use the native .join(' ') to concat it to a readable string.

File: server/db.js [hunk 4]
```diff
@@ -14,85 +14,94 @@ const config = {
...
-    id: id,
-    user_icon: 'google.com',
-    review_text: generateRandomReview(100),
+    userIcon: 'google.com',
+    reviewText: generateRandomReview(100),
     date: new Date(),
-    name: random_names[getRandomInt(0, random_names.length)],
+    name: randomNames[getRandomInt(0, randomNames.length)],
     rating: {
       cleanliness: getRandomInt(1, 5),
```
Comment (chrisalopez1337) [inline server/db.js]: Ratings shouldnt be prime int's they can be floats (1.56) you could consider changing your random function to generate a string and concat on the decimal (random(1,5) + '.' + random(1,99)); Or think of a different way to accomplish this.

**Review by chrisalopez1337 — CHANGES_REQUESTED** [review]
Commented on lines where we found some changes to be made

File: server/db.js [hunk 5]
```diff
@@ -14,12 +14,19 @@ const config = {
...
 };
 
-// get random number between min and max, inclusive
+// get random int between min and max, inclusive
 const getRandomInt = (min, max) => {
   let range = max - min;
   return Math.floor(Math.random() * range) + min;
 }
 
+// get random float between min and max, inclusive and rounded to numDecimalPlaces
```
Comment (chrsbell) [inline server/db.js:23]: Added getRandomFloat method to round Number to 2 decimal places

File: server/db.js [hunk 6]
```diff
@@ -31,15 +38,16 @@ const randomText = ['apple', 'orange', 'pear', 'little', 'big', 'hack', 'reactor
...
-    let index = Math.floor(Math.random() * randomText.length);
-    // console.log(index);
-    review += randomText[index] + ' ';
+const generateRandomReview = () => {
+  let review = [];
+  // generate random review of length between 50-150
+  let reviewLength = getRandomInt(50,150);
+  for (let i = 0; i < reviewLength; i++) {
+    // get a random word from randomText
+    let index = getRandomInt(0, randomText.length);
```
Comment (chrsbell) [inline server/db.js:47]: Fixed

File: server/db.js (hunk 4 above)
Comment (chrsbell) [reply server/db.js]: https://github.com/keybox130/Reviews-Service/pull/1/files/0b9b2ebdee1f09bca326960678f7fc69a9b5ba3c..014a120f3f4739003f7d084f60d9fe05b115aaec#r495199422

File: server/db.js [hunk 7]
```diff
@@ -31,15 +38,16 @@ const randomText = ['apple', 'orange', 'pear', 'little', 'big', 'hack', 'reactor
...
-const generateRandomReview = (length) => {
-  let review = '';
-  for (let i = 0; i < length; i++) {
-    let index = Math.floor(Math.random() * randomText.length);
-    // console.log(index);
-    review += randomText[index] + ' ';
+const generateRandomReview = () => {
+  let review = [];
+  // generate random review of length between 50-150
+  let reviewLength = getRandomInt(50,150);
```
Comment (chrsbell) [inline server/db.js:44]: Generate length of review randomly inside the function

File: server/db.js (hunk 2 above)
Comment (chrsbell) [reply server/db.js]: https://github.com/keybox130/Reviews-Service/pull/1/files/0b9b2ebdee1f09bca326960678f7fc69a9b5ba3c..014a120f3f4739003f7d084f60d9fe05b115aaec#r495200218

File: server/db.js [hunk 8]
```diff
@@ -78,30 +86,40 @@ const reviewSchema = new mongoose.Schema({
...
-      console.log('Successfully initialized database');
-    })
-    .catch(err => {
-      console.log(err);
-    });
+  // populate the database once we are connected (if necessary!)
+  Review.find({}).exec()
+  .then(docs => {
+    // check if there are already docs in the database
+    if (!docs.length) {
```
Comment (chrsbell) [inline server/db.js:113]: Only seed the database if there are no documents already

File: server/db.js (hunk 1 above)
Comment (chrsbell) [reply server/db.js:37]: https://github.com/keybox130/Reviews-Service/pull/1/files/0b9b2ebdee1f09bca326960678f7fc69a9b5ba3c..014a120f3f4739003f7d084f60d9fe05b115aaec#r495200408

File: server/db.js [hunk 9]
```diff
@@ -31,15 +38,16 @@ const randomText = ['apple', 'orange', 'pear', 'little', 'big', 'hack', 'reactor
...
+  // generate random review of length between 50-150
+  let reviewLength = getRandomInt(50,150);
+  for (let i = 0; i < reviewLength; i++) {
+    // get a random word from randomText
+    let index = getRandomInt(0, randomText.length);
+    review.push(randomText[index]);
   }
-  review = review.slice(0, review.length - 1);
-  return review;
+  return review.join(' ');
```
Comment (chrsbell) [inline server/db.js:50]: Use the native .join string method

File: server/db.js (hunk 3 above)
Comment (chrsbell) [reply server/db.js]: https://github.com/keybox130/Reviews-Service/pull/1/files/0b9b2ebdee1f09bca326960678f7fc69a9b5ba3c..014a120f3f4739003f7d084f60d9fe05b115aaec#r495200620

**Review by chrisalopez1337 — APPROVED** [review]
Everything looks great! Pulling to master.

**Pull request merged**